### PR TITLE
425-refactor: WidgetTitle and MainTitle refactor

### LINKED ISSUE #425: FSD: Widget-title and Main-title refactor
## 📝 Describe the refactoring task

- add `data-testid`,
- `margin-bottom` --> `padding-bottom`,
- make `Widget-title` more unified - do not pass paddings in the props.

## 📍 Code location

`src/shared/ui/main-title`,
`src/shared/ui/widget-title`

## ✅ Definition of Done (DoD):

Everything described has been done

## 🕰️ Timeline:

16 hours

### PR DESCRIPTION
**Pull request by ansivgit**
## What type of PR is this? (select all that apply)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [x] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description

- added `data-testid`,
- added `padding-bottom` depending on the title size,
- removed unused styles,
- removed `size="medium"` from default variant of `WidgetTitle` component,
- Playwright snapshots updated.

## Related Tickets & Documents

- Related Issue #425 
- Closes #425 

## Screenshots, Recordings

_Please replace this line with any relevant images for UI changes._

## Added/updated tests?

- [x] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

## Summary by CodeRabbit

- **New Features**
	- Improved styling for widget titles with updated spacing and class names.
- **Bug Fixes**
	- Enhanced visibility assertions for the `MainTitle` and `WidgetTitle` components in tests.
- **Chores**
	- Removed redundant `size` props from various components to simplify usage and maintain consistency in styling.

### CODE REVIEW

**Review by KristiBo — APPROVED** [review]

File: src/shared/ui/widget-title/widget-title.tsx [hunk 1]
```diff
@@ -9,33 +9,33 @@ type WidgetTitleProps = HTMLAttributes<HTMLHeadingElement>
...
+      asterisk: cx('asterisk'),
+    },
   },
   defaultVariants: {
     size: 'medium',
     mods: null,
   },
 });
 
-export const WidgetTitle = ({ children, size, mods, className, ...props }: WidgetTitleProps) => {
```
Comment (SpaNb4) [inline src/shared/ui/widget-title/widget-title.tsx:30]: Same as https://github.com/rolling-scopes/site/pull/419#discussion_r1699504907

File: src/shared/ui/widget-title/widget-title.module.scss [hunk 2]
```diff
@@ -9,29 +9,34 @@
...
   @include media-tablet {
+    padding-bottom: 16px;
     font-size: 36px;
     line-height: 44px;
   }
+
+  padding-bottom: 20px;
+  font-size: 44px;
+  line-height: 52px;
 }
```
Comment (SpaNb4) [inline src/shared/ui/widget-title/widget-title.module.scss:28]: I'd prefer to have css properties listed first, and only then media queries
```
.medium {
  padding-bottom: 20px;
  font-size: 44px;
  line-height: 52px;

  @include media-tablet {
    padding-bottom: 16px;
    font-size: 36px;
    line-height: 44px;
  }
}
```
Let's discuss this with the team.

File: src/shared/ui/widget-title/widget-title.tsx (hunk 1 above)
Comment (SpaNb4) [reply src/shared/ui/widget-title/widget-title.tsx:30]: I mean `type WidgetTitleProps = VariantProps<typeof widgetTitleVariants>;`

Comment (ansivgit) [reply src/shared/ui/widget-title/widget-title.tsx:30]: We've an error here 
image

Comment (SpaNb4) [reply src/shared/ui/widget-title/widget-title.tsx:30]: I see two ways to fix this
```
type WidgetTitleProps = Pick<HTMLAttributes<HTMLHeadingElement>, 'className' | 'children'> &
  VariantProps<typeof widgetTitleVariants>;
```

```
type WidgetTitleProps = { children: React.ReactNode;
  className?: string; } & VariantProps<
  typeof widgetTitleVariants
  >;
```

**Review by SpaNb4 — APPROVED** [review]

**Pull request merged**